python/lsst/ip/isr/photodiodeCorrection.py

```diff
@@ -0,0 +1,205 @@
+        RuntimeError :
+            Raised if the supplied dictionary is for a different
+            calibration type.
+        """
+        calib = cls()
+
+        if calib._OBSTYPE != dictionary['metadata']['OBSTYPE']:
+            raise RuntimeError(f"Incorrect photodiode correctionsupplied.  Expected {calib._OBSTYPE}, "
+                               f"found {dictionary['metadata']['OBSTYPE']}")
+
+        calib.setMetadata(dictionary['metadata'])
+        for pair in dictionary['pairs']:
+            correction = dictionary['pairs'][pair]
+            calib.abscissaCorrections[pair] = correction
+
+        calib.updateMetadata()
+
+        return calib
+
+    def toDict(self):
+        """Return a dictionary containing the photodiode correction properties.
+
+        The dictionary should be able to be round-tripped through.
+        `fromDict`.
+
+        Returns
+        -------
+        dictionary : `dict`
+            Dictionary of properties.
+        """
+        self.updateMetadata()
+
+        outDict = {}
+        outDict['metadata'] = self.getMetadata()
+        for pair in self.abscissaCorrections.keys():
+            outDict['pairs'][pair] = self.abscissaCorrections[pair]
+
+        return outDict
+
+    @classmethod
+    def fromTable(cls, tableList):
+        """Construct calibration from a list of tables.
+
+        This method uses the `fromDict` method to create the
+        calibration after constructing an appropriate dictionary from
+        the input tables.
+
+        Parameters
+        ----------
+        tableList : `list` [`astropy.table.Table`]
+            List of tables to use to construct the crosstalk
+            calibration.
+
+        Returns
+        -------
+        calib : `lsst.ip.isr.PhotodiodeCorrection`
+            The calibration defined in the tables.
+        """
+        dataTable = tableList[0]
+
+        metadata = dataTable.meta
+        inDict = dict()
+        inDict['metadata'] = metadata
+        inDict['pairs'] = dict()
+
+        for record in dataTable:
+            pair = record['PAIR']
+            inDict['pairs'][pair] = record['PD_CORR']
+
+        return cls().fromDict(inDict)
+
+    def toTable(self):
+        """Construct a list of tables containing the information in this
+        calibration.
+
+        The list of tables should create an identical calibration
+        after being passed to this class's fromTable method.
+
+        Returns
+        -------
+        tableList : `list` [`astropy.table.Table`]
+            List of tables containing the photodiode correction
+            information.
+        """
+        tableList = []
+        self.updateMetadata()
+        catalog = Table([{'PAIR': key,
+                          'PD_CORR': self.abscissaCorrections[key]}
+                         for key in self.abscissaCorrections.keys()])
+        catalog.meta = self.getMetadata().toDict()
+        tableList.append(catalog)
+
+        if self.tableData is not None:
```

Comment:
I don't think these `LOOKUP_VALUES` are being persisted correctly, are they?  They appear here, but I don't see an associated block in `fromTable` or in `to/fromDict`.

Can you add a test (`tests/test_photodiode.py` is probably the least confusing place) that at least does round-tripping tests?

Reply:
I will work on this next week (week of Feb 14).

Reply:
You were right, those weren't being persisted.  I fixed this in several places and created a tests/test_photodiodeCorrection.py with round trip tests, which now pass.